src/test/java/bio/terra/service/filedata/google/gcs/GcsPdaoTest.java

```diff
@@ -70,6 +78,86 @@ public void testGetBlobWithHash() {
     }
   }
 
+  @Test
+  public void testGetBlobContentsMatchingPath() {
+    String uuid = UUID.randomUUID().toString();
+    List<BlobId> blobIds =
+        List.of(
+            BlobId.of(testConfig.getIngestbucket(), uuid + "/" + uuid + "-1.txt"),
+            BlobId.of(testConfig.getIngestbucket(), uuid + "/" + uuid + "-2.txt"),
+            BlobId.of(testConfig.getIngestbucket(), uuid + "/" + uuid + "-3.txt"));
+
+    try {
+      List<String> contents = new ArrayList<>();
+      for (int i = 0; i < blobIds.size(); i++) {
+        var fileContents = String.format("This is line %d", i);
+        var blobId = blobIds.get(i);
+        storage.create(
+            BlobInfo.newBuilder(blobId).build(), fileContents.getBytes(StandardCharsets.UTF_8));
+        contents.add(fileContents);
+      }
+      var listPath = GcsPdao.getGsPathFromComponents(testConfig.getIngestbucket(), uuid + "/");
+      var listLines = gcsPdao.getGcsFilesLines(listPath, projectId);
+      assertThat(
+          "The listed file contents match concatenated contents of individual files",
+          listLines,
+          equalTo(contents));
+
+      var wildcardMiddlePath =
+          GcsPdao.getGsPathFromComponents(
+              testConfig.getIngestbucket(), uuid + "/" + uuid + "-*.txt");
+      var wildcardMiddleLines = gcsPdao.getGcsFilesLines(wildcardMiddlePath, projectId);
+      assertThat(
+          "The middle-wildcard-matched file contents match concatenated contents of individual files",
+          wildcardMiddleLines,
+          equalTo(contents));
+
+      var wildcardEndPath =
+          GcsPdao.getGsPathFromComponents(testConfig.getIngestbucket(), uuid + "/" + uuid + "-*");
+      var wildcardEndLines = gcsPdao.getGcsFilesLines(wildcardEndPath, projectId);
+      assertThat(
+          "The end-wildcard-matched file contents match concatenated contents of individual files",
+          wildcardEndLines,
+          equalTo(contents));
+
+      var wildcardMultiplePath =
+          GcsPdao.getGsPathFromComponents(
+              testConfig.getIngestbucket(), uuid + "/*" + uuid + "-*.txt");
+      var wildcardMutlipleLines = gcsPdao.getGcsFilesLines(wildcardMultiplePath, projectId);
+      assertThat(
+          "Multiple-wildcard paths don't return anything",
+          wildcardMutlipleLines,
+          emptyCollectionOf(String.class));
+
+    } finally {
+      storage.delete(blobIds);
+    }
+  }
+
+  @Test
+  public void testWriteLinesToNewGcsFile() {
```

Comment:
As long as the new  `GcsPdao` code is exercised by an end-to-end test, it would be better to write these unit tests by mocking `gcsProjectFactory` in `GcsPdao`, to avoid creating more connected tests. To get the full benefit they would have to be in a unit test class instead of a connected test one.

Reply:
Are we not supposed to be creating more connected tests?

Reply:
I would say yes, we don't want to create any more connected tests unless there's no other way to test it. I think `testGetBlobContentsMatchingPath()` could be moved to a non-connected test. Not sure about `testWriteLinesToNewGcsFile()`, as this is testing a thin wrapper around GCS operations. If this code is already tested elsewhere due to its use in a larger test case, then it would be safe to convert this to a mocked test as well.

Reply:
I've removed this test. Due to the creation of `GcsIO` and it being tested every time we test ingest, I think its fair to remove this test.